### DISCUSSION TITLE
Call init() function twice

### DISCUSSION THREAD

**Discussion by vchinaski** [i18next/i18next#1753] [Q&A]
Hi everyone! I have a question about odd behavior I experienced in my project. We have our wrapper `init()` function, it looks like that:
```
async init(version: number, resources?: Resource) {
        const backendOpts = {
            backend: {
                loadPath: `/static/i18n/${version}/{{lng}}/{{ns}}.json`
            }
        };

        const resourceOpts = resources ? { resources } : backendOpts;

        await this.i18n.init({
            ns: this.sources,
            defaultNS: this.defaultSource,
            // English will be always our first fallback language if a translation for specified language doesn't exist
            fallbackLng: SUPPORTED_LNGS,
            supportedLngs: SUPPORTED_LNGS,
            load: "currentOnly",
            detection: {
                caches: ["localStorage"]
            },
            // we don't need to do that since it's done by default in React
            interpolation: {
                escapeValue: false
            },
            ...resourceOpts
        });

        this.isLoaded = true;
    }
```

As you see, we use `i18next-http-backend` library for loading resources. If you just do `await init(<some-version>)`, everything works as expected. But if you call this function twice (basically, call `i18n.init()` twice), all keys are missing and `ResourceStore` `this.data` is empty. I think this is a culprit: https://github.com/i18next/i18next/blob/3c757ab15c284fae55298f4bd7cb712b9a0c0d89/src/i18next.js#L102
Basically, when you run `init()` for the first time, default `data` is `{}` (since we don't use `resources`) but later the store is hydrated by fetching the resource from the backend. But when you do `init()` second time, the store is flashed because of the line above, but there's no re-hydration happens. I'm curios why it's working like that?

Thank you in advance for your response!

**Comment by vchinaski** [comment]
Any help, please? 🙏

**Reply by jamuhl** [reply]
don't call init twice...it really should not be needed to call init multiple times

**Reply by vchinaski** [reply]
@jamuhl I know, but when you do it accidentally, `i18n` backend feature will be broken and you'll have no idea why :( Your store will become empty with no real reason or error message.

**Reply by jamuhl** [reply]
because you can init it only once...be it by accident or not...init it only once...it just does not allow to call init multiple times...
